python/understack-workflows/understack_workflows/bmc_chassis_info.py

```diff
@@ -184,11 +206,21 @@ def interface_detail(bmc, path) -> dict:
     InterfaceEnabled, LinkStatus, Status.Health, State.Enabled, SpeedMbps
     """
     data = bmc.redfish_request(path)
+    name = [value for key, value in data.items()
+            if 'name' in key.lower()]
+    hostname = [value for key, value in data.items()
+                if 'hostname' in key.lower()]
+    description = [value for key, value in data.items()
+                   if 'description' in key.lower()]
+    mac_addrs = [value for key, value in data.items()
+                 if 'macaddress' in key.lower()]
+    mac_addr = mac_addrs[0] if len(mac_addrs) > 0 else ''
```

Comment:
Consider simplifying this by normalizing the data on input:

```suggestion
    data = {k.lower(): v for k, v in data.items()}
    name = data.get('name')
    hostname = data.get('hostname', '')
    description = data.get('description')
    mac_addr = data.get('macaddress', '')
```
(note that these are now all strings, not lists)